### PR TITLE
Fix bug with calculation of partially correct scores

### PR DESCRIPTION
**Pull request by taylortom**

### CODE REVIEW

**Comment by hrajotia** [comment]
+1

File: js/adapt-contrib-mcq.js [hunk 1]
```diff
@@ -166,20 +166,16 @@ define(function(require) {
...
+            this.model.set('_numberOfRequiredAnswers', numberOfRequiredAnswers);
 
             // Check if correct answers matches correct items and there are no incorrect selections
-            if (numberOfCorrectAnswers === numberOfCorrectItems && numberOfIncorrectAnswers === 0) {
-                return true;
-            } else {
-                return false;
-            }
-            
+            var answeredCorrectly = numberOfCorrectAnswers === numberOfRequiredAnswers && numberOfIncorrectAnswers === 0;
```
Comment (darylhedley) [inline js/adapt-contrib-mcq.js]: Could we have this written as ``(numberOfCorrectAnswers === numberOfRequiredAnswers) && (numberOfIncorrectAnswers === 0);`` - makes it easier to read as your evaluating

**Comment by darylhedley** [comment]
Does this not still give you 6.6666666667 as your _score if you have`` _questionWeight:10, and two out of three items correct``? I'm still not sure if you get it half right whether you should get any _score

**Comment by moloko** [comment]
I agree - partially correct answers would only require a score in *very* exceptional cases. maybe we could just drop this altogether and if someone needs it they could just implement by hand? wouldn't be hard to do. 


—
Matt Leathes

**Comment by darylhedley** [comment]
@taylortom - could we push this fix to the master branch too please as it's a hotfix?

**Comment by taylortom** [comment]
See https://github.com/adaptlearning/adapt-contrib-mcq/issues/76

What do we reckon's the best solution? I don't have a preference re. including the partial score functionality, but definitely think it should be disabled by default.

**Comment by oliverfoster** [comment]
It'd be nice to have a default _allowPartialScoring: false on the questionView prototype.

**Comment by moloko** [comment]
sounds good to me, though the name is not 100% clear - maybe
'_scorePartiallyCorrectAnswers'?

On 5 January 2015 at 09:34, Oliver Foster <notifications@github.com> wrote:

> It'd be nice to have a default _allowPartialScoring: false on the
> questionView prototype.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/adaptlearning/adapt-contrib-mcq/pull/84#issuecomment-68685297>
> .
>



-- 
--------------------------------------------------------------------------------------------------------------------------------
IMPORTANT: This email is intended for the use of the individual
addressee(s) named above and may contain information that is confidential,
privileged, or unsuitable for overly sensitive persons with low
self-esteem, no sense of humour, or irrational religious beliefs.
If you have received this email in error, please add some egg whites and
nutmeg, whisk and place in a warm oven for 40 minutes.

**Comment by oliverfoster** [comment]
We know what i'm like with naming.... x) 

Perhaps _canScorePartiallyCorrectAnswers   ? inline with _canShowFeedback

**Comment by moloko** [comment]
+1

On 5 January 2015 at 09:56, Oliver Foster <notifications@github.com> wrote:

> We know what i'm like with naming.... x)
>
> Perhaps _canScorePartiallyCorrectAnswers ? inline with _canShowFeedback
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/adaptlearning/adapt-contrib-mcq/pull/84#issuecomment-68687483>
> .
>



-- 
--------------------------------------------------------------------------------------------------------------------------------
IMPORTANT: This email is intended for the use of the individual
addressee(s) named above and may contain information that is confidential,
privileged, or unsuitable for overly sensitive persons with low
self-esteem, no sense of humour, or irrational religious beliefs.
If you have received this email in error, please add some egg whites and
nutmeg, whisk and place in a warm oven for 40 minutes.

**Comment by taylortom** [comment]
Haven't added rounding, as I'm not sure what's best; I'm leaning towards keeping the accuracy, and just leaving the dev to deal with any rounding. Thoughts?

**Comment by oliverfoster** [comment]
I thought the idea was to allow it to be either correct or incorrect, scored or unscored.

i.e.     score = (answeredCorrectly ? 0 : 1) * questionWeight

**Comment by oliverfoster** [comment]
+1

**Comment by moloko** [comment]
+1

**Comment by zenduo** [comment]
+1

**Comment by brian-learningpool** [comment]
+1, can we get this patched on master too?

**Comment by taylortom** [comment]
Yep, will sort that now

**Pull request merged**